flink-runtime/src/test/java/org/apache/flink/runtime/jobmaster/DefaultJobMasterServiceProcessTest.java

```diff
@@ -70,6 +78,69 @@ public void testInitializationFailureCompletesResultFuture() {
         assertThat(initializationFailure, containsCause(originalCause));
     }
 
+    @Test
+    public void testInitializationFailureSetsFailureInfoProperly()
+            throws ExecutionException, InterruptedException {
+        final CompletableFuture<JobMasterService> jobMasterServiceFuture =
+                new CompletableFuture<>();
+        DefaultJobMasterServiceProcess serviceProcess = createTestInstance(jobMasterServiceFuture);
+        final RuntimeException originalCause = new RuntimeException("Expected RuntimeException");
+
+        long beforeFailureTimestamp = System.currentTimeMillis();
+        jobMasterServiceFuture.completeExceptionally(originalCause);
+        long afterFailureTimestamp = System.currentTimeMillis();
+
+        final JobManagerRunnerResult result = serviceProcess.getResultFuture().get();
+        final ErrorInfo executionGraphFailure =
+                result.getExecutionGraphInfo().getArchivedExecutionGraph().getFailureInfo();
+
+        assertNotNull(executionGraphFailure);
+        final Throwable actualException =
+                executionGraphFailure
+                        .getException()
+                        .deserializeError(Thread.currentThread().getContextClassLoader());
+        assertEquals(JobInitializationException.class, actualException.getClass());
+        assertTrue(
+                ExceptionUtils.findThrowable(actualException, t -> t.equals(originalCause))
+                        .isPresent());
+
+        assertTrue(executionGraphFailure.getTimestamp() >= beforeFailureTimestamp);
+        assertTrue(executionGraphFailure.getTimestamp() <= afterFailureTimestamp);
+    }
+
+    @Test
+    public void testInitializationFailureSetsExceptionHistoryProperly()
```

Comment:
These new two tests are more or less the same. Should we either merge them or have an abstract test that we reuse?

Reply:
Good point, I moved the assertions into a helper method. 👍